x/auth/client/rest/query.go

```diff
@@ -197,16 +205,27 @@ func packStdTxResponse(w http.ResponseWriter, clientCtx client.Context, txRes *s
 	return nil
 }
 
-func checkSignModeError(w http.ResponseWriter, ctx client.Context, resp interface{}, grpcEndPoint string) error {
+// checkSignModeError checks if there are errors with marshalling proto-only
+// txs with amino.
+func checkSignModeError(ctx client.Context, resp interface{}, grpcEndPoint string) error {
 	// LegacyAmino used intentionally here to handle the SignMode errors
 	marshaler := ctx.LegacyAmino
 
 	_, err := marshaler.MarshalJSON(resp)
-	if err != nil && strings.Contains(err.Error(), unRegisteredConcreteTypeErr) {
-		rest.WriteErrorResponse(w, http.StatusInternalServerError,
-			"This transaction was created with the new SIGN_MODE_DIRECT signing method, and therefore cannot be displayed"+
-				" via legacy REST handlers, please use CLI or directly query the Tendermint RPC endpoint to query"+
-				" this transaction. gRPC gateway endpoint is "+grpcEndPoint)
+	if err != nil {
+		isKnownError := false
+		for _, knownError := range knownErrors {
+			if strings.Contains(err.Error(), knownError) {
+				isKnownError = true
+			}
+		}
```

Comment:
I think we have a Tendermint function we can use for this. We call it in one or two places in the codebase -- I can't recall the function name...

Reply:
I tried to find all occurrences of string.contains in the tendermint repo, didn't find anything.